src/services/AuthService.ts

```diff
@@ -0,0 +1,59 @@
+import { Role } from "../models";
+import { User } from "../models";
+import { ErrorWrongData } from "../errors";
+import bcript from "bcryptjs";
+import { UserDB } from "../types";
+import { prepareUserObject, generateAccessToken } from "../helpers";
+import { lengthPassword } from "../constants";
+
+export class AuthService {
+  static async registration(username: string, password: string) {
+    try {
+      const res = await User.findOne({ username });
+
+      if (res) throw new ErrorWrongData("User already exists");
+
+      const hashPassword = bcript.hashSync(password, lengthPassword);
+      const userRole = await Role.findOne({ value: "USER" });
+
+      if (!userRole) throw new ErrorWrongData("Unabled to create user");
+
+      const user = new User({
+        username,
+        password: hashPassword,
+      });
+      await user.save();
+
+      return { message: "User successfully created" };
+    } catch (error) {
+      return { message: error.message };
+    }
+  }
+
+  static async login(username: string, password: string) {
+    const users: Array<UserDB> = await User.find({ username });
```

Comment:
in this method we can have an error throwed too. Lets wrap it on try catch and other methods too